src/lib/util/Games/WheelOfFortune.ts

```diff
@@ -0,0 +1,192 @@
+		return (this.message.guild && guildBoosts.includes(this.message.guild.id) ? 1.5 : 1)
+			* (userBoosts.includes(this.message.author.id) ? 1.5 : 1);
+	}
+
+	public async run() {
+		const { settings } = this.player;
+		const money = settings.get(UserSettings.Money);
+		const darkTheme = settings.get(UserSettings.DarkTheme);
+		const spin = Math.floor(Math.random() * this.MULTIPLIERS.length);
+		this.calculate(spin);
+
+		const amount = this.winnings === 0 ? money - this.amount : money - this.amount + (this.winnings * this.boost);
+		if (amount < 0) throw this.message.language.tget('GAMES_CANNOT_HAVE_NEGATIVE_MONEY');
+		await settings.update(UserSettings.Money, amount);
+		return {
+			attachment: await this.render(money, spin, amount, darkTheme),
+			amount
+		};
+	}
+
+	private calculate(spin: number) {
+		// The multiplier to apply
+		const multiplier = this.MULTIPLIERS[spin];
+
+		// The winnings
+		this.winnings += this.amount * multiplier;
+	}
+
+	private async render(money: number, spin: number, amount: number, darkTheme: boolean): Promise<Buffer> {
+		const playerHasWon = amount > money;
+		const darkColour = '#2F3136';
+		const lightColour = '#FFFFFF';
+
+		const { x: arrowX, y: arrowY } = this.getArrow(kArrows.values(), spin)!;
+
+		const canvas = new Canvas(300, 120)
+			.save()
+			.setShadowColor('rgba(51, 51, 51, 0.38)')
+			.setShadowBlur(5)
+			.setColor(darkTheme ? darkColour : lightColour)
+			.createBeveledClip(4, 4, 300 - 8, 142, 5)
+			.fill()
+			.restore()
+			.save()
+			.setColor(darkTheme ? lightColour : darkColour)
+			.setTextFont('30px RobotoLight')
+			.setTextAlign('right')
+			.addText(this.message.language.tget('COMMAND_WHEELOFFORTUNE_CANVAS_TEXT', playerHasWon), 280, 60)
+			.addText(playerHasWon ? (this.winnings - this.amount).toString() : (this.winnings + this.amount).toString(), 250, 100)
+			.addImage(WheelOfFortune.images.SHINY!, 260, 68, 20, 39)
+			.addImage(WheelOfFortune.images.ARROWS!, arrowX, arrowY, kArrowSize, kArrowSize, kIconSize + 4, kIconSize + 4, kIconSize, kIconSize)
+			.restore();
+
+		await Promise.all(kAssets.map(value => new Promise(resolve => {
+			const { x, y } = value;
+			canvas.addImage(
+				playerHasWon ? WheelOfFortune.images.WIN_ICONS! : WheelOfFortune.images.LOSE_ICONS!,
+				x, y, kIconSize, kIconSize, x + 4, y + 4, kIconSize, kIconSize
+			);
+			resolve();
+		})));
+
+		return canvas.toBufferAsync();
+	}
+
+	private getArrow(iterable: IterableIterator<Coordinate>, n: number) {
+		let index = 0;
+		for (const arrow of iterable) {
+			if (index === n) {
+				return arrow;
+			}
+			index++;
+		}
+		return null;
+	}
+
+	public static images = Object.seal<WheelOfFortuneAssets>({
+		LOSE_ICONS: null,
+		WIN_ICONS: null,
+		ARROWS: null,
+		SHINY: null
+	});
+
+	public static async init(): Promise<void> {
+		const [winIcons, loseIcons, arrows, shiny] = await Promise.all([
+			loadImage(join(assetsFolder, 'images', 'social', 'wof-win-icons.png')),
```

Comment:
Can't you join the path to the social assets in a constant, and use that?

Reply:
Ye I suppose so